src/scenario/scenario_event.c

```diff
@@ -0,0 +1,196 @@
+    new_condition->type = condition->type;
+    new_condition->parameter1 = condition->parameter1;
+    new_condition->parameter2 = condition->parameter2;
+    new_condition->parameter3 = condition->parameter3;
+    new_condition->parameter4 = condition->parameter4;
+    new_condition->parameter5 = condition->parameter5;
+}
+
+scenario_action_t *scenario_event_get_action(scenario_event_t *event, int id)
+{
+    return array_item(event->actions, id);
+}
+
+scenario_action_t *scenario_event_action_create(scenario_event_t *event, int type)
+{
+    scenario_action_t *action = 0;
+    array_new_item(event->actions, 0, action);
+    if (!action) {
+        return 0;
+    }
+    action->type = type;
+
+    return action;
+}
+
+void scenario_event_link_action(scenario_event_t *event, scenario_action_t *action)
+{
+    scenario_action_t *new_action = 0;
+    array_new_item(event->actions, 0, new_action);
+
+    new_action->type = action->type;
+    new_action->parameter1 = action->parameter1;
+    new_action->parameter2 = action->parameter2;
+    new_action->parameter3 = action->parameter3;
+    new_action->parameter4 = action->parameter4;
+    new_action->parameter5 = action->parameter5;
+}
+
+void initialize_new_scenario_event(scenario_event_t *event, int position)
+{
+    event->id = position;
+}
+
+int scenario_event_is_valid(const scenario_event_t *event)
+{
+    return event->state != EVENT_STATE_UNDEFINED;
+}
+
+
+int scenario_event_can_repeat(scenario_event_t *event)
+{
+    return (event->repeat_months_min > 0)
+        && (event->repeat_months_max >= event->repeat_months_min);
+}
+
+int scenario_event_all_conditions_met(scenario_event_t *event)
+{
+    if (event->state != EVENT_STATE_ACTIVE) {
+        return 0;
+    }
+    if (event->actions.size < 1) {
+        return 0;
+    }
+    
+    int passed = 1;
+
+    for (int i = 0; i < event->conditions.size; i++) {
+        scenario_condition_t *current = array_item(event->conditions, i);
+        int condition_check = scenario_condition_is_met(current);
+        passed &= condition_check;
+    }
+
+    return passed;
+}
+
+int scenario_event_decrease_pause_time(scenario_event_t *event, int months_passed)
+{
+    if (event->state != EVENT_STATE_PAUSED) {
+        return 0;
+    }
+
+    if (event->months_until_active > 0) {
+        event->months_until_active -= months_passed;
+    }
+    if (event->months_until_active < 0) {
+        event->months_until_active = 0;
+    }
+    if (event->months_until_active == 0) {
+        event->state = EVENT_STATE_ACTIVE;
+    }
+    return 0;
+}
+
+int scenario_event_conditional_execute(scenario_event_t *event)
+{
+    if (scenario_event_all_conditions_met(event))
+    {
+        int result = scenario_event_execute(event);
+        if (scenario_event_can_repeat(event)) {
+            event->state = EVENT_STATE_PAUSED;
+            event->months_until_active = random_between_from_stdlib(event->repeat_months_min, event->repeat_months_max);
```

Comment:
Reminder: Add rerun of init for event so that any random rolls (conditions + actions) can be redone.